### ISSUE TITLE
Segmentation fault with condition option

### ISSUE THREAD

**Issue by jgmayer67** [zhanxw/rvtests#48]
Hi,

I'm new to RVTest and genetics, so if this is something simple I apologize in advance.  I am trying to run a conditional analysis with one SNP as the covariate.  My command line is below, as well as the error message.  This runs fine without the condition option.  When I add the condition option, I get a few seconds of it running and then the following occurs:

Include sample [ 49679653 ].
./condtest.sh: line 40: 22942 Segmentation fault      (core dumped) $RVTEST --inVcf $VCFIN --pheno $PEDIN --dosage DS --covar $PEDIN --pheno-name $PHEN_NAME1 --covar-name $COV_NAME --condition $COND_RANGE --peopleIncludeFile $COMP_LIST --out $ASSCOUT1 --single $STAT_TEST

There are 12333 samples, and the log file has 24736 lines of output, so it appears to have gotten into some processing before the error.   I have also experimented with widening the range on the condition, same result.

Any help you can provide would be appreciated.

Thanks,

John 
  
$RVTEST --inVcf $VCFIN --pheno $PEDIN --dosage DS  --covar $PEDIN --pheno-name $PHEN_NAME1 --covar-name $COV_NAME --condition 6:32632605-32632605  --peopleIncludeFile $COMP_LIST --out $ASSCOUT1 --single $STAT_TEST

**Comment by zhanxw** [comment]
Did this segfault happen only when "--condition" is used?

Can you paste some lines from the log file?

Thanks.

**Comment by jgmayer67** [comment]
Yes, if I remove the condition argument the program runs fine.  Here are some of the beginning output lines:

Effective Options
    --inVcf chr6.dose.emerge_ids.consented.merged.HC.vcf.gz
    --out chr6.CondAnalysisTestCondion
    --covar HC_RVTEST_V3_FOR_CA.ped
    --covar-name erh,site,PC1,PC2,PC3
    --pheno COV_PHEN_DAT_EM_GWAS_HC_RVTEST_V3_FOR_CA.ped
    --pheno-name SCORE
    --dosage DS
    --peopleIncludeFile GOOD_BATCH.list
    --single wald
    --condition 6:32632605-32632605

[INFO]  Program version: 20170418
[INFO]  Analysis started at: Thu Nov 30 07:50:18 2017
Include sample [ 63002450 ].
Include sample [ 63008167 ].
Include sample [ 63010203 ].

about halfway through the log file, after all the include samples I get:

[INFO]  Loaded [ 12333 ] samples from VCF files
[INFO]  Loaded [ 13783 ] sample pheontypes
[INFO]  Discard [ 1450 ] samples as they do not have genotypes
[INFO]  Begin to read covariate file
[WARN]  Total [ 1450 ] samples are skipped from covariate file due to missing phenotype
[WARN]  Skip sample [ 16218931 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16219298 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16219621 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16219983 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16221028 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16221790 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16221903 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16222144 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16223659 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16224888 ] from covariate file due to missing phenotype
[WARN]  Skip sample [ 16225310 ] from covariate file due to missing phenotype
[WARN]  Skip outputting additional [ 1440 ] samples from covariate file with missing phenotypes
[INFO]  Loaded 5797 male, 6535 female and 1 sex-unknonw samples from

Hope that helps.

Thanks,

John

**Comment by limjyvictor** [comment]
Hi,

I wonder if there is any update on this issue?
I encounter the same problem when trying to do conditional analysis.

Thanks!
Victor

**Comment by nhalliga** [comment]
I am having the same problem.  I'm using rvtests on a server and requesting 64GB of memory.  I want to condition on a single variant, so it's not clear how big a region to give --condition.  The output/error file results follows (edited for brevity and to remove identifiers):

The following parameters are available.  Ones with "[]" are in effect:

Available Options
      Basic Input/Output:
                          --inVcf [/MY_DATA]
                          --inBgen [], --inBgenSample [], --inKgg []
                          --out [MY_results_conditioned_on_1SNP]
                          --outputRaw
       Specify Covariate: --covar [covar.txt], --covar-name [c1,c2,c3,c4,c5,c6,c7,c8,c9,c10], --sex
       Specify Phenotype: --pheno [Pheno.txt], --inverseNormal
                          --useResidualAsPhenotype, --mpheno [], --pheno-name []
                          --qtl, --multiplePheno []
        Specify Genotype: --dosage [DS], --multipleAllele
    Chromosome X Options: --xLabel [], --xParRegion []
           People Filter: --peopleIncludeID [], --peopleIncludeFile []
                          --peopleExcludeID [], --peopleExcludeFile []
             Site Filter: --rangeList [], --rangeFile [], --siteFile []
                          --siteDepthMin [], --siteDepthMax [], --siteMACMin []
                          --annoType []
         Genotype Filter: --indvDepthMin [], --indvDepthMax [], --indvQualMin []
       Association Model: --single [wald,score], --burden [], --vt []
                          --kernel [], --meta []
     Family-based Models: --kinship [], --xHemiKinship [], --kinshipEigen []
                          --xHemiKinshipEigen [], --boltPlink []
                          --boltPlinkNoCheck
          Grouping Unit : --geneFile [], --gene [], --setList [], --setFile []
                          --set []
        Frequency Cutoff: --freqUpper [], --freqLower []
            Missing Data: --impute [], --imputePheno, --imputeCov
    Conditional Analysis: --condition [11:5303063-5303064]
     Auxiliary Functions: --noweb, --hide-covar, --numThread [], --outputID
                          --help

Effective Options
    --inVcf /MY_DATA
    --out MY_results_conditioned_on_1SNP
    --covar covar.txt
    --covar-name c1,c2,c3,c4,c5,c6,c7,c8,c9,c10
    --pheno Pheno.txt
    --dosage DS
    --single wald,score
    --condition 11:5303063-5303064

[INFO]	Program version: 20171009
[INFO]	Analysis started at: Mon Dec 17 10:57:43 2018
[INFO]	Loaded [ 1076 ] samples from genotype files
[INFO]	Loaded [ 1044 ] sample phenotypes

[INFO]	Loaded 538 male, 506 female and 0 sex-unknown samples from rvvPheno.txt
Include sample [ A BUNCH OF SAMPLES ].



/var/spool/torque/mom_priv/jobs/32387356.nyx.arc-ts.umich.edu.SC: line 19: 18495 Segmentation fault      (core dumped) rvtest --inVcf /MY_DATA --pheno Pheno.txt --covar covar.txt --covar-name c1,c2,c3,c4,c5,c6,c7,c8,c9,c10 --dosage DS --condition 11:5303063-5303064 --out MY_results_conditioned_on_1SNP --single wald,score

Note that there is a segmentation fault.  The command to analyze the data worked without --condtion.  This also failed with --condition 11:5303063-5303063 and --condition 11:5303063.  I really only want to condition on a single SNP in an imputed dataset.

This is the log file from rvtests:
[INFO]	Program version: 20171009
[INFO]	Git Version: 5a2e770cdc1cc583febd2a5555b65624e595e7bf
[INFO]	Parameters BEGIN
# ParameterList created by nhalliga on nyx6152.arc-ts.umich.edu at Mon Dec 17 10:57:43 2018
--inVcf "/MY_DATA" --out "MY_results_conditioned_on_1SNP" --covar "covar.txt" --covar-name "c1,c2,c3,c4,c5,c6,c7,c8,c9,c10" --pheno "Pheno.txt" --dosage "DS" --single "wald,score" --condition "11:5303063-5303064"
[INFO]	Parameters END

Thanks.
